packages/nx/src/tasks-runner/pseudo-terminal.ts

```diff
@@ -155,18 +172,20 @@ export class PseudoTtyProcess {
   }
 
   onOutput(callback: (message: string) => void): void {
-    this.childProcess.onOutput(callback);
+    this.outputCallbacks.push(callback);
   }
 
   kill(): void {
-    try {
-      this.childProcess.kill();
-    } catch {
-      // when the child process completes before we explicitly call kill, this will throw
-      // do nothing
-    } finally {
-      if (this.isAlive == true) {
-        this.isAlive = false;
+    if (this.isAlive) {
+      try {
+        this.childProcess.kill();
+      } catch {
+        // when the child process completes before we explicitly call kill, this will throw
+        // do nothing
+      } finally {
+        if (this.isAlive == true) {
+          this.isAlive = false;
+        }
```

Comment:
This can probably be simplified to just this since we check if its alive above:
```suggestion
      } finally {
        this.isAlive = false;  
```